pkg/probe/exec/exec.go

```diff
@@ -39,7 +46,52 @@ type execProber struct{}
 // from executing a command. Returns the Result status, command output, and
 // errors if any.
 func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
-	data, err := e.CombinedOutput()
+	stdoutPipe, err := e.StdoutPipe()
```

Comment:
I started a PR for this with adding a LimitWriter utility: https://github.com/tallclair/kubernetes/commit/843ef55d715735d2672f020305ef4771ddaf55fd

I think you can just set `e.Stdout` and `e.Stderr` to a limit writer, and simplify a lot of the following logic.

Reply:
Note that setting them to the same writer will interleave the output too. This is all that `CombinedOutput` is doing under the hood (https://golang.org/src/os/exec/exec.go?s=15900:15946#L541)